frontend/src/app/VersionEnforcer.tsx

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from "react";
+import { RouteComponentProps, withRouter } from "react-router";
+
+import { VersionService } from "./VersionService";
+
+const VersionEnforcer: React.FunctionComponent<RouteComponentProps> = (
+  props: RouteComponentProps
+) => {
+  useEffect(() => {
+    (async () => {
+      const remoteSHA = await VersionService.getSHA();
+      if (process.env.REACT_APP_CURRENT_COMMIT !== remoteSHA) {
```

Comment:
How's our app frontend app caching currently? I could see this being an awful infinite loop if the app is cached but commit.txt has a newer sha

Reply:
The main app bundle, where that environment variable would be encoded at build time, is cache-busted with a hash in the filename

Reply:
Sorry, I should have been more specific, I was thinking about the index.html file caching. I think we should add a cache control header to the index.html file to make sure _that_ isn't cached because getting the same index.html file as before will result in the pointing to the old bundles https://create-react-app.dev/docs/production-build/#static-file-caching